src/modules/ekf2/EKF/aid_sources/external_vision/ev_vel_control.cpp

```diff
@@ -75,80 +77,81 @@ void Ekf::controlEvVelFusion(const extVisionSample &ev_sample, const bool common
 	case VelocityFrame::LOCAL_FRAME_FRD:
 		if (_control_status.flags.ev_yaw) {
 			// using EV frame
-			vel = ev_sample.vel - vel_offset_earth;
-			vel_cov = matrix::diag(ev_sample.velocity_var);
+			measurement = ev_sample.vel - vel_offset_earth;
+			measurement_var = ev_sample.velocity_var;
 
 		} else {
 			// rotate EV to the EKF reference frame
 			const Dcmf R_ev_to_ekf = Dcmf(_ev_q_error_filt.getState());
 
-			vel = R_ev_to_ekf * ev_sample.vel - vel_offset_earth;
-			vel_cov = R_ev_to_ekf * matrix::diag(ev_sample.velocity_var) * R_ev_to_ekf.transpose();
-
-			// increase minimum variance to include EV orientation variance
-			// TODO: do this properly
-			const float orientation_var_max = ev_sample.orientation_var.max();
-
-			for (int i = 0; i < 2; i++) {
-				vel_cov(i, i) = math::max(vel_cov(i, i), orientation_var_max);
-			}
+			measurement = R_ev_to_ekf * ev_sample.vel - vel_offset_earth;
+			measurement_var = matrix::SquareMatrix3f(R_ev_to_ekf * matrix::diag(ev_sample.velocity_var) *
+					  R_ev_to_ekf.transpose()).diag();
```

Comment:
You probably copy-pasted that equation several times, please extract it to a helper function